src/scipp/core/variable.py

```diff
@@ -782,7 +792,23 @@ def arange(dim: str,
                                              start=start,
                                              stop=stop,
                                              step=step)
-    return array(dims=[dim], values=_np.arange(**range_args), unit=unit, dtype=dtype)
+    args = [x for x in [start, stop, step] if x is not None]
+    types = [
+        x.values.dtype if isinstance(x, _cpp.Variable) else _np.asarray(x).dtype
+        for x in args
+    ]
```

Comment:
Shouldn't handling None and different types be dealt with by `_normalize_range_args`?

Reply:
We can't do what we need to do on the output of `_normalize_range_args`.